avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
You can raise generic exceptions from libraries. The most appropriate here would be the `IOError` as it is in fact Intup/Output error.

What is not recommended from libraries is to raise `avocado.core.exceptions.Test*` exceptions, because they are kind-of special. The difference is that untreated IOError in test reports as ERROR. In most cases this is correct as you do not intend to test writing to the file. But if you were to develop a test to check whether writing certain value to that file works, you'd have to try/catch this exception and raise `self.fail` (or simply used `fail_on` decorator to say IOError is expected failure). Then the test status would be FAIL (on exception).

Is this clearer now, or should I expand the examples?